examples/sharing_asset.py

```diff
@@ -0,0 +1,191 @@
+        auth_token=getenv(f"ARCHIVIST_AUTHTOKEN_{label}"),
+    )
+    # Initialize connection to Archivist. max_time is the time to wait for confirmation
+    # of an asset or event creation - the default is 1200 seconds but one can optionally
+    # specify a different value here particularly when creating assets on SIMPLE_HASH
+    # (rather than KHIPU) as confirmation times are much shorter in this case.
+    return Archivist(
+        "https://app.rkvst.io",
+        auth,
+        max_time=300,
+    )
+
+
+def exchange_subject(acme, weyland):
+    """Add subjects record for weyland in acme's environment"""
+    subject = acme.subjects.exchange(
+        "weyland",
+        weyland.subjects.read(SUBJECTS_SELF_ID),
+    )
+
+    # must wait for confirmation
+    acme.subjects.wait_for_confirmation(subject["identity"])
+
+    return subject
+
+
+def create_access_policy(arch, label, subject):
+    """Create access policy"""
+    # consists of a filter selection entry and a selection criteria to restrict/redact
+    # values of the asset attributes available to the sharee.
+
+    # values pertaining to the access polciy itself.
+    props = {
+        "display_name": f"{label} access policy",
+        "description": f"{label} Policy description",
+    }
+
+    # Filtering - access will be allowed to any asset that contains both these
+    # attributes that equal these values. This happens to match the asset created
+    # previously.
+    filters = [
+        {
+            "or": [
+                f"attributes.arc_display_type={label}_display_type",
+            ]
+        },
+        {
+            "or": [
+                f"attributes.ext_vendor_name={label}",
+            ]
+        },
+    ]
+
+    # one must be the subject to gain access and only those fields
+    # specified in include_attributes will be emitted.
+    access_permissions = [
+        {
+            "subjects": [
+                subject["identity"],
+            ],
+            "behaviours": ASSET_BEHAVIOURS,
+            "include_attributes": [
+                "arc_display_name",
+            ],
+        },
+    ]
+
+    return arch.access_policies.create(
+        props,
+        filters,
+        access_permissions,
+    )
+
+
+def main():
+    """Main function of share-asset."""
+    # optional call to set the logger level for all subsystems. The argument can
+    # be either "INFO" or "DEBUG". For more sophisticated logging control see the
+    # documentation.
+    set_logger("INFO")
+
+    # For demonstration purposes we are going to assume that 2 organizations are
+    # going to share an asset. The 2 organizations are ACME Corp and Weyland-Yutani
+    # Corporation.
+    acme = create_archivist("acme")
+    weyland = create_archivist("weyland")
+
+    # set a subject for weyland in acme's environment. The identity will be used as a
+    # filter in the access permissions of the access_policy.
+    weyland_subject = exchange_subject(acme, weyland)
```

Comment:
This variable name is slightly ambiguous - It could be misinterpreted as a subject within the weyland tenancy. It might be easier to follow subsequent code by naming this `weyland_subject_on_acme` or similar.